### ISSUE TITLE
ICLR '18 | Don't Decay the Learning Rate, Increase the Batch Size

### ISSUE THREAD

**Issue by vycezhong** [vycezhong/read-papers#8]
https://arxiv.org/pdf/1711.00489.pdf

**Comment by vycezhong** [comment]
很有意思的paper. 基于一个别人的理论基础。这个理论引入了一个叫做noise scale的概念，这是控制random抖动幅度的量。其中eps是学习率，m是momentum的系数，N是total samples，B是batch size。由于N >> B，所以 N/B - 1 约等于 N/B. 

![image](https://user-images.githubusercontent.com/25879526/92301766-09b34600-ef99-11ea-99c7-f8e09212fcd8.png)

这样显然可以推导出 #4 得到的Linear Scaling Rule的结论

这篇文章提出的idea是，一般来讲会做decay learning rate，根据linear scaling rule，这等同于增大batch size。然后他们实验发现确实是这样，这么做没有问题. 

实际上很少有人会这么做，因为如果使用cosine decay呢？你每个step都调batch size？另一方面，为什么不一开始就把batch size增大到显存极限呢？其实文章这个idea在实际中，我没看到有任何人使用。

另一个结论是可以增加momentum系数，来增大batch size. 但是实验结果表明会有一点accuracy drop.
![image](https://user-images.githubusercontent.com/25879526/92301832-8cd49c00-ef99-11ea-93f1-f712925bd5f5.png)
 

说实话noise scale这个概念很有趣，有时间看下推导

**Comment by vycezhong** [comment]
关于解释增大momentum为什么不work那段证明，我觉得有点扯。。。

**Comment by vycezhong** [comment]
https://arxiv.org/pdf/1710.06451.pdf

证明这个用到了随机微分方程orz，实在看不懂

**Comment by vycezhong** [comment]
这篇其实有应用. #102  如果考虑elastic training，运行时增加worker，就能增加batch size.

**Comment by vycezhong** [comment]
突然发现这篇讲得挺好的. 
![image](https://user-images.githubusercontent.com/25879526/115139584-441fae00-a065-11eb-9219-ffc914f60573.png)

这个公式其实点明了SGD的几件事情:

1. linear scaling rule为什么成立.

2. 为什么大batch不成立（因为不满足 B << N）

3. 在满足B<<N的时候，降lr和加batch效果等同. 

缺点:

1. 没有考虑momentum的noise scale. #177  

同时考虑gradient和momentum noise scale就是一个比较完整的解决方案了. #177
